module/modmanager/manager.go

```diff
@@ -417,6 +418,67 @@ func (mgr *Manager) ValidateConfig(ctx context.Context, conf resource.Config) ([
 	return resp.Dependencies, nil
 }
 
+// ResolveImplicitDependenciesInConfig modifies the config diff to add implicit dependencies to changed components
+// and updates modular components & services whose module has been changed with new implicit deps and adds them to conf.Modified.
```

Comment:
```suggestion
// ResolveImplicitDependenciesInConfig modifies the config diff to add implicit dependencies to changed resources
// and updates modular resources whose module has been changed with new implicit deps and adds them to conf.Modified.
```
Took me a second to understand this comment, but LGTM. Just want to clarify we're doing the same thing for both components and services (resources).